src/forms/create_ticket.ts

```diff
@@ -20,3 +31,199 @@ export async function newCreateTicketForm(call: AppCall): Promise<AppForm> {
+                return;
+            }
+
+            // only keep fields listed in formIDs
+            if (formIDs.includes(field.id)) {
+                fields.push(field);
+            }
+        });
+        return fields;
+    }
+
+    mapZdFieldsToAppFields(fields: Users.Fields.UserField[]): void {
+        fields.forEach((field: Users.Fields.UserField) => {
+            const name = this.getMappedName(field);
+            const label = field.title;
+            const isRequired = Boolean(field.required_in_portal);
+
+            switch (field.type) {
+            case ZDFieldTypes.Description:
+                // will be filled by post message and handled separately
+                return;
+
+            case ZDFieldTypes.Integer:
+            case ZDFieldTypes.Decimal:
+            case ZDFieldTypes.Subject:
+            case ZDFieldTypes.Text:
+            case ZDFieldTypes.MultiLine: {
+                const f = new TextField(name);
+                f.setLabel(label);
+                if (isRequired) {
+                    f.isRequired();
+                }
+                if (field.type === ZDFieldTypes.MultiLine) {
+                    f.isTextArea();
+                }
+                this.builder.addField(f.toAppField());
+                return;
+            }
+
+            case ZDFieldTypes.Checkbox: {
+                const f = new BoolField(name);
+                f.setLabel(label);
+                if (isRequired) {
+                    f.isRequired();
+                }
+                this.builder.addField(f.toAppField());
+                return;
+            }
+
+            case ZDFieldTypes.TicketType:
+            case ZDFieldTypes.Priority:
+            case ZDFieldTypes.Tagger:
+            case ZDFieldTypes.Muliselect: {
+                const options = this.isSystemField(field) ? field.system_field_options : field.custom_field_options;
+                const f = new StaticSelectField(name, makeOptions(options));
+                f.setLabel(label);
+                if (isRequired) {
+                    f.isRequired();
+                }
+                if (field.type === ZDFieldTypes.Muliselect) {
+                    f.isMuliselect();
+                }
+                this.builder.addField(f.toAppField());
+                return;
+            }
+            default:
+                console.log('field type not mapped to app field. type =', field.type);
+                break;
+            }
+        });
+    }
+
+    isSystemField(field: Users.Fields.UserField): boolean {
+        return (SystemFields.includes(String(field.type)) || field.system_field_options);
+    }
+
+    // getMappedName gets the mapped field name for a zendesk field.
+    // custom zendesk field names are prefixed so that it can be easily parsed
+    // when form is submitted
+    getMappedName(field: Users.Fields.UserField): string {
+        const strFieldType = String(field.type);
+        switch (true) {
```

Comment:
why not a sequence of `if...else` s?

Reply:
done. I was expecting more cases, but did not turn out that way.